app/utilities/PersistenceManager.java

```diff
@@ -87,6 +88,23 @@ public static Token getToken(String token){
     }
 
     /**
+     * Persists restaurant into database.
+     *
+     * @param restaurant the Restaurant to be persisted
+     */
+    public static void saveRestaurant(Restaurant restaurant){
+        restaurant.getAddress().setRestaurantId(Restaurant.find.all().size() + 1);
+        restaurant.getAddress().setEmail(restaurant.getName().replace(" ","").toLowerCase() + "@restauranttest.com");
```

Comment:
There was a "structure" issue regarding Address model class which forced me to add email to restaurants (which isn't needed at all). Now I solved that and fixed the hardcoded things. Won't happen again :)